src/tim/fuse/batch_fuse.cc

```diff
@@ -0,0 +1,558 @@
+
+#include "tim/fuse/batch_fuse.h"
+// #include "builtin_op_impl.h"
+#include "batch_fuse_context.h"
+#include <algorithm>
+#include <deque>
+#include <cassert>
+
+#include "tim/vx/context.h"
+#include "tim/vx/graph.h"
+#include "tim/vx/operation.h"
+
+#include "ops/conv2d_batch_fuse.h"
+// #include "ops/pad_v2_batch_fuse.h"
+#include "ops/pad_batch_fuse.h"
+#include "ops/activation_batch_fuse.h"
+#include "ops/elementwise_batch_fuse.h"
+#include "ops/pool2d_batch_fuse.h"
+#include "ops/reduce_batch_fuse.h"
+#include "ops/transpose_batch_fuse.h"
+#include "ops/reshape_batch_fuse.h"
+#include "ops/concat_batch_fuse.h"
+
+namespace tim {
+namespace fuse {
+namespace batch_fuse_impl {
+
+std::vector<std::shared_ptr<vx::Tensor>> HandleBatchFuse(
+    std::shared_ptr<batch_fuse_impl::BatchFuseContext>& ctx,
+    const std::shared_ptr<vx::Operation>& op);
+
+void BatchFuseContext::MarkVisited(const std::shared_ptr<vx::Operation>& op) {
+  if (visited_op_.end() !=
+      std::find(visited_op_.begin(), visited_op_.end(), op)) {
+    VSILOGW("The operation has been mark as visited.");
+  } else {
+    visited_op_.push_back(op);
+  }
+}
+
+bool BatchFuseContext::IsVisited(
+    const std::shared_ptr<vx::Operation>& op) const {
+  if (visited_op_.end() !=
+      std::find(visited_op_.begin(), visited_op_.end(), op)) {
+    return true;
+  } else {
+    return false;
+  }
```

Comment:
return (visited_op_.end() != std::find( ...)) ; instead of if-else.

Reply:
Done